scripts/migrate-yaml.py

```diff
@@ -0,0 +1,164 @@
+    return parser.parse_args()
+
+
+def root_cleanup(var):
+    """Cleans up what deepdiff returns so we can have some cleaner output.
+
+    :param var: the key you wish to clean up, just a string
+    :return: the cleaned up string
+    """
+    prefix = 'root[\''
+    suffix = '\']'
+    var = var[var.startswith(prefix) and len(prefix):]
+    var = var[:var.endswith(suffix) and -len(suffix)]
+    var = var.replace('\'][\'', '.')
+    return var
+
+
+def get_config(path):
+    """Load yaml data if you can
+
+    :param path: path of yaml data to load
+    :return: dictionary
+    """
+    try:
+        with open(path) as f:
+            data = f.read()
+    except IOError as e:
+        if e.errno == errno.ENOENT:
+            data = None
+        else:
+            raise e
+    else:
+        data = yaml.safe_load(data)
+
+    if data is None:
+        return {}
+    else:
+        return data
+
+
+def do_the_diff(defaults, overrides):
+    """Diff two yaml files, returning four dictionaries containing cleaned up key-value sets.
+
+    :param defaults: File you wish to compare against, the base.
+    :param overrides: File you wish to compare, the mask
+    :return: four dictionaries, what is added, what was taken away and what was changed.
+    """
+    added = {}
+    removed = {}
+    changed = {}
+    changed_types = {}
+
+    diff = DeepDiff(defaults, overrides, verbose_level=2)
+
+    if 'dictionary_item_added' in diff.keys():
+        for old_key in diff['dictionary_item_added'].keys():
+            added[root_cleanup(old_key)] = diff['dictionary_item_added'].pop(old_key)
+    if 'dictionary_item_removed' in diff.keys():
+        for old_key in diff['dictionary_item_removed'].keys():
+            removed[root_cleanup(old_key)] = diff['dictionary_item_removed'].pop(old_key)
+    if 'values_changed' in diff.keys():
+        for old_key in diff['values_changed'].keys():
+            changed[root_cleanup(old_key)] = diff['values_changed'].pop(old_key)
+    if 'type_changes' in diff.keys():
+        for old_key in diff['type_changes'].keys():
+            changed_types[root_cleanup(old_key)] = diff['type_changes'].pop(old_key)
+
+    # have to recast the values as standard dictionaries and combine type and value changes...
+    new_changed = {}
+    new_changed_types = {}
+    for key in changed:
+        new_changed[key] = dict(changed[key])
+    for key in changed_types:
+        new_changed_types[key] = dict(changed_types[key])
+    new_changed.update(new_changed_types)
+
+    # we make two different yaml dump-able dictionaries, one for new values, one for old (what is being overridden)
+    newdict = {}
+    olddict = {}
+    for key in new_changed:
+        newdict[key] = new_changed[key]['new_value']
+        olddict[key] = new_changed[key]['old_value']
+
+    return dict(added), dict(removed), dict(newdict), dict(olddict)
+
+if __name__ == '__main__':
+    # boilerplate goes here
+    args = parse_args()
```

Comment:
This section of code is *untestable* unless it's in one (or preferably more) functions.